fetch/stale-while-revalidate/resources/stale-script.py

```diff
@@ -1,10 +1,13 @@
-import random, string, datetime
+import random, string, datetime, time
 
 def id_token():
    letters = string.ascii_lowercase
    return ''.join(random.choice(letters) for i in range(20))
 
 def main(request, response):
+    is_revalidation = request.headers.get("If-None-Match", None)
+    if is_revalidation is not None:
+      time.sleep(5);
```

Comment:
I would add a parameter in the URL to control whether the revalidation is hanging.